types/unist/index.d.ts

```diff
@@ -6,13 +6,21 @@
 //                 Titus Wormer <https://github.com/wooorm>
 //                 Junyoung Choi <https://github.com/rokt33r>
 //                 Ben Moon <https://github.com/GuiltyDolphin>
+//                 JounQin <https://github.com/JounQin>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.0
 
 /**
  * Syntactic units in unist syntax trees are called nodes.
  */
-export interface Node {
+export interface Node<
+    /**
+     * Information associated by the ecosystem with the node.
+     * Space is guaranteed to never be specified by unist or specifications
+     * implementing unist.
+     */
+    Data = unknown,
```

Comment:
But `Data` is not `unknown`, it’s supposed to be a `Record<string, unknown>` 🤔 Why change that?

Reply:
OK

Reply:
To @wooorm's point, I believe `Data` is generally supposed to be object like
```suggestion
    Data extends { [key: string]: unknown } = { [key: string]: unknown },
```

example of a similar interface https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAUwB4EMC2AHANsgHgBF0p0VUpkwATAZ0QG8BtAa2QE8AuROqAJxhgA5gF0e4VmDgB3MAF9EAXiZtOPPoJHjEk6XPkA+ABQ1S6HiTIBKS+aYAoRIn7IoIfkjNkH8hw7QsPGRjRjo4TGQABX44bB4BEGR5ayA

Reply:
![image](https://user-images.githubusercontent.com/8336744/125122719-64bd4880-e128-11eb-91f4-12b8b22c8eec.png)

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAUwB4EMC2AHANsgHgBF0p0VUpkwATAZ0QG8BtAa2QE8AuROqAJxhgA5gF0e4VmDgB3MAF9EAXiZtOPPoJHjEk6XPkA+ABQ1S6HiTIBKS+aYAoRIn7IoIfkjNkH8hw7RsOH4oRCEqfmB0CGREAEFHZ3ZuXgEhYV9-NCw8QjiTRjo4TGQABX44bB4BEGR5ayA

`extends object` should be OK

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAUwB4EMC2AHANsgHgBF0p0VUpkwATAZ0TgCMArZaRAXkQG8BtANbIAngC5EdKACcYYAOYBdceAFg4AdzABfAHwAKGqXTiSZAJQmjvAFCJEU5FBBSkhsta3XrabHClREWSopYHQIZEQAQRs7ITEJaVk5Dy80LDxCSP0eOjhMZAAFKThscWkQZC0zIA

Reply:
:thinking: what about:
https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAUwB4EMC2AHANsgHgChFEBpZATwGcVUpkwATW6qAJxjAHNEBeRG048ANCUQARdFHR0GzWgG8A2gGsqtLuQ0BdAFyJwqsHADuYAL79EK9TURaKNfYbDGzlogD4AFE2noBlIyAJRBATbi7MhQIOxI-jJEFkREaNhw7FAOYAzswOgQyIgAgpGkdgZCXNzJqWhYeIR2cMClIqW+itRwmMgACuxw2AYcIMgWIUA

Reply:
Is there any downside by `extends object`? It's working. Less generic is better for this case.

Reply:
`Object` type also includes literals, like `string`
https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAUwB4EMC2AHANsgHgChFEARdKdFVKZMAEwGdEB5AIwCtlpEBeRAG8A2gGtkATyYAuREygAnGGADmAXVnhRYOAHcwAXyIA+ABQNK6WRSoBKa5aElEC5FBAKkFqkSNEiaFh4hPJKqmYA5N7UTAAWcCC4DIjouLroUojsyC48cArJuDDiADRZIFCIMExgADoRlbHIrhG2QA

Reply:
`object` is not `Object`